File: Coding skills/parkBill.cpp
//The problem: https://app.codility.com/programmers/trainings/5/parking_bill/
#include <iostream>

using namespace std;

int solution(string &E, string &L)
{
    int counter{};

    int hourE = stoi(E);
    int minutesE = stoi(E.substr(3));
    int hourL = stoi(L);
    int minutesL= stoi(L.substr(3));
    int totalHours = hourL - hourE;
    int totalMinutes =  minutesL- minutesE;
    if (totalHours >0)
    {
        counter =1;
        if(totalMinutes>0)
        {
            totalHours +=1;
        };
        counter += totalHours *4;
    }
    else{
        if(totalMinutes>0)
        {
            counter =5;
        }
    }
return counter;
};

int main()
{
    string E = "09:00";
    string L = "09:30";
    cout<<solution(E,L)<<endl;;
}

File: Coding skills/park_bill.py
# The problem: https://app.codility.com/programmers/trainings/5/parking_bill/
import math


def parking_sum(E, L):
    # copy the times without the colon.
    newE, newL = E[:2] + E[3:], L[:2] + L[3:]
    # calculate the time
    total_hours = (int(newL) - int(newE)) / 100
    # round upwards
    round_total = math.ceil(total_hours)
    # elegant way to calculate the bill
    finel = round_total * 4 + 1
    return finel


if __name__ == '__main__':
    print(parking_sum("10:00", "13:38"))
